src/services/WaveformDataUtils.js

```diff
@@ -259,40 +303,30 @@ export default class WaveformDataUtils {
   /**
    * Prevent the times of segment being edited overlapping with the existing segments
    * @param {Object} segment - segement being edited in the waveform
+   * @param {Boolean} inMarker - true -> start time changed, false -> end time changed
```

Comment:
I see that this is the new variable you've added throughout the code.  Your description is clear, but I'm confused as to why it is called `inMarker`.  I wonder if a different name might be clearer?  Maybe even `startTimeChanged`?

Reply:
Yeah, I used the same name they used in Peaks.js. I agree, it doesn't make sense in here. :)